mbf_abstract_nav/src/controller_action.cpp

```diff
@@ -0,0 +1,267 @@
+#include "mbf_abstract_nav/controller_action.h"
+
+namespace mbf_abstract_nav{
+
+
+ControllerAction::ControllerAction(
+    const std::string &action_name,
+    const RobotInformation &robot_info)
+    : AbstractAction(action_name, robot_info, boost::bind(&mbf_abstract_nav::ControllerAction::run, this, _1, _2))
+{
+}
+
+
+void ControllerAction::before(GoalHandle &goal_handle, AbstractControllerExecution& execution)
+{
+  if(threads_.size() == 1)
```

Comment:
Should not be >= 1? If not, only first call to controller works, as threads_ keeps increasing size with every call